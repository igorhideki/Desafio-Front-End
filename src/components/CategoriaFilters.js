import React from 'react';
import PropTypes from 'prop-types';
import CategoriaFiltersItem from '../components/CategoriaFiltersItem';

const CategoriaFilters = (props) => (
  <div className="filter">
    <label className="filter__label">Categorias</label>
    {
      props.categorias && props.categorias.map((categoria) => {
        return (
          <CategoriaFiltersItem 
            key={categoria.type}
            name={categoria.name}
            type={categoria.type}
            filterActive={props.filterActive}
            handleSelectCategoria={props.handleSelectCategoria}
          />
        );
      })
    }
  </div>
);

CategoriaFilters.propTypes = {
  categorias: PropTypes.arrayOf(PropTypes.shape({
    type: PropTypes.number,
    name: PropTypes.string
  })),
  handleSelectCategoria: PropTypes.func,
  filterActive: PropTypes.number
};

export default CategoriaFilters;
