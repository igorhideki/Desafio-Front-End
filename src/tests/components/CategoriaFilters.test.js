import React from 'react';
import { shallow } from 'enzyme';
import CategoriaFilters from '../../components/CategoriaFilters';
import categorias from '../fixtures/categorias';

test('should render CategoriasFilters correctly', () => {
  const wrapper = shallow(<CategoriaFilters categorias={categorias}/>);
  expect(wrapper).toMatchSnapshot();
});
