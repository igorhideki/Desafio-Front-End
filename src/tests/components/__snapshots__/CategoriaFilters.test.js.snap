// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CategoriasFilters correctly 1`] = `
<div
  className="filter"
>
  <label
    className="filter__label"
  >
    Categorias
  </label>
  <CategoriaFiltersItem
    key="0"
    name="Informativos SUN"
    type={0}
  />
  <CategoriaFiltersItem
    key="1"
    name="ForÃ§a de Vendas"
    type={1}
  />
  <CategoriaFiltersItem
    key="2"
    name="Vivo Ligado"
    type={2}
  />
  <CategoriaFiltersItem
    key="3"
    name="Turbine Suas Vendas"
    type={3}
  />
</div>
`;
